MonoGame.Framework/Circle.cs

```diff
@@ -0,0 +1,509 @@
+		/// </summary>
+		public bool IsEmpty
+		{
+			get
+			{
+				return ((this.Radius == 0) && (this.Center == Vector2.Zero));
+			}
+		}
+
+		#endregion
+
+		#region Internal Properties
+
+		internal string DebugDisplayString
+		{
+			get
+			{
+				return string.Concat(
+					this.Center.ToString(), "  ",
+					this.Radius
+					);
+			}
+		}
+
+		#endregion
+
+		#region Constructors
+
+		/// <summary>
+		/// Creates a new instance of <see cref="Circle"/> struct, with the specified
+		/// position, and radius
+		/// </summary>
+		/// <param name="center">The position of the center of the created <see cref="Circle"/>.</param>
+		/// <param name="radius">The radius of the created <see cref="Circle"/>.</param>
+		public Circle(Vector2 center, float radius)
+		{
+			this.Center = center;
+			this.Radius = radius;
+		}
+
+		#endregion
+
+		#region Operators
+
+		/// <summary>
+		/// Compares whether two <see cref="Circle"/> instances are equal.
+		/// </summary>
+		/// <param name="a"><see cref="Circle"/> instance on the left of the equal sign.</param>
+		/// <param name="b"><see cref="Circle"/> instance on the right of the equal sign.</param>
+		/// <returns><c>true</c> if the instances are equal; <c>false</c> otherwise.</returns>
+		public static bool operator ==(Circle a, Circle b)
+		{
+			return ((a.Center == b.Center) && (a.Radius == b.Radius));
+		}
+
+		/// <summary>
+		/// Compares whether two <see cref="Circle"/> instances are not equal.
+		/// </summary>
+		/// <param name="a"><see cref="Circle"/> instance on the left of the not equal sign.</param>
+		/// <param name="b"><see cref="Circle"/> instance on the right of the not equal sign.</param>
+		/// <returns><c>true</c> if the instances are not equal; <c>false</c> otherwise.</returns>
+		public static bool operator !=(Circle a, Circle b)
+		{
+			return !(a == b);
+		}
+
+		#endregion
+
+		#region Public Methods
+
+		/// <summary>
+		/// Gets the point at the edge of this <see cref="Circle"/> from the provided angle
+		/// </summary>
+		/// <param name="angle">an angle in radians</param>
+		/// <returns><see cref="Vector2"/> representing the point on this <see cref="Circle"/>'s surface at the specified angle</returns>
+		public Vector2 GetPointAlongEdge(float angle)
+		{
+			return new Vector2(Center.X + (Radius * (float)Math.Cos((double)angle)),
+							   Center.Y + (Radius * (float)Math.Sin((double)angle)));
+		}
+
+
+		/// <summary>
+		/// Gets whether or not the provided coordinates lie within the bounds of this <see cref="Circle"/>.
+		/// </summary>
+		/// <param name="x">The x coordinate of the point to check for containment.</param>
+		/// <param name="y">The y coordinate of the point to check for containment.</param>
+		/// <returns><c>true</c> if the provided coordinates lie inside this <see cref="Circle"/>; <c>false</c> otherwise.</returns>
+		public bool Contains(float x, float y)
+		{
+			return ((new Vector2(x, y) - Center).Length() <= Radius);
```

Comment:
`Length()` does a square root internally, which is inefficient.  It is more common to do something like

    return ((new Vector2(x, y) - Center).LengthSquared() <= Radius * Radius);